backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/KonanIrlinker.kt

```diff
@@ -186,30 +185,12 @@ internal class KonanIrLinker(
                 moduleDescriptor.findClassAcrossModuleDependencies(classId) ?: error("No declaration found with $idSig")
             }
 
-        private fun getIrFile(packageFragment: PackageFragmentDescriptor): IrFile {
-            val fqn = packageFragment.fqName
-            return packageToFileMap.getOrPut(packageFragment.fqName) {
-                val fileSymbol = IrFileSymbolImpl(packageFragment)
-                IrFileImpl(NaiveSourceBasedFileEntryImpl("forward declarations for $fqn"), fileSymbol, packageFragment.fqName).also {
-                    moduleFragment.files.add(it)
-                }
+        private fun buildForwardDeclarationStub(descriptor: ClassDescriptor): IrClass {
+            return stubGenerator.generateClassStub(descriptor).also {
```

Comment:
I generally have no idea what is the difference here, so my approve of this patch doesn't mean that I've actually verified it.

Reply:
The difference is following. Previously we constructed IR class for descriptor "by hand" and forgot to translate its supertypes (which caused the bug). Now we use Lazy IR mechanism here which automatically translates supertypes. 
This change was checked by @romanart.
As a little positive side-effect, forward declarations are removed from `context.irModule` which saves us a couple of milliseconds during compile-time :)

Reply:
I understand this part. What I don't understand is why didn't we use Lazy IR mechanism before?

Reply:
Probably because we didn't use it before IR Linker refactoring (when `buildForwardDeclarationStub` logic was moved from common part).